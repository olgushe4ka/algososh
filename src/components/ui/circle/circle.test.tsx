import React from "react";
import { render } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle", () => {
  it("renders correctly without letter", () => {
    const { container } = render(<Circle />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with letter", () => {
    const { container } = render(<Circle letter="A" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with head", () => {
    const { container } = render(<Circle head="H" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with React element in head", () => {
    const { container } = render(
      <Circle head={<span className="test">Test</span>} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with tail", () => {
    const { container } = render(<Circle tail="T" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with React element in tail", () => {
    const { container } = render(
      <Circle tail={<span className="test">Test</span>} tailType="element" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with index", () => {
    const { container } = render(<Circle index={1} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with isSmall prop", () => {
    const { container } = render(<Circle isSmall={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly in default state", () => {
    const { container } = render(<Circle />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly in changing state", () => {
    const { container } = render(<Circle state={ElementStates.Changing} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly in modified state", () => {
    const { container } = render(<Circle state={ElementStates.Modified} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});