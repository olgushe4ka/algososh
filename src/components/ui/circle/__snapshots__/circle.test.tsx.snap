// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders correctly in changing state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly in default state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly in modified state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with React element in head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <span
      class="test"
    >
      Test
    </span>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with React element in tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <span
      class="test"
    >
      Test
    </span>
  </div>
</div>
`;

exports[`Circle renders correctly with head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    H
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with index 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle renders correctly with isSmall prop 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    T
  </div>
</div>
`;

exports[`Circle renders correctly without letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render correctly with head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render correctly with letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render correctly with react element in head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      head
    </div>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle should render correctly with react element in tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      tail
    </div>
  </div>
</div>
`;

exports[`Circle should render correctly with tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle should render correctly without letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
